### PR TITLE
Be stricter in typescript build for tree dds

### PR DESCRIPTION
**Pull request by CraigMacomber**
Ban implicit any, and use strict null checks in tests.

### CODE REVIEW

File: packages/dds/tree/package.json [hunk 1]
```diff
@@ -73,6 +73,7 @@
     "@types/mocha": "^8.2.2",
     "@types/node": "^14.18.0",
     "@types/random-js": "^1.0.31",
+    "@types/ungap__structured-clone": "^0.3.0",
```
Comment (yann-achard-MS) [inline packages/dds/tree/package.json:76]: Is this change related to the PR's purpose?

**Review by DLehenbauer — APPROVED** [review]
Pre-approved.

**Review by yann-achard-MS — CHANGES_REQUESTED** [review]

File: packages/dds/tree/package.json (hunk 1 above)
Comment (CraigMacomber) [reply packages/dds/tree/package.json:76]: Yes. This is necessary to cause packages/dds/tree/src/utils.ts to not implicitly use the any type for the import of structured-clone.

**Review by yann-achard-MS — APPROVED** [review]
